tests/ducktape/producer_strategy.py

```diff
@@ -67,40 +80,134 @@ def _print_timing_metrics(self, producer_type, produce_times, poll_times, flush_
         print(f"Total poll() calls: {len(poll_times)}")
         print(separator)
 
+    def build_serializers(self, serialization_type, is_async=False):
+        """Build and return (key_serializer, value_serializer) for the given serialization_type.
+        Returns (None, None) if serialization_type is not provided.
+
+        Args:
+            serialization_type: Type of serialization ('avro', 'json', 'protobuf')
+            is_async: Whether to use async serializers (for AsyncProducerStrategy)
+        """
+        if not serialization_type:
+            return None, None
+        schema_registry_url = os.getenv(
+            'SCHEMA_REGISTRY_URL',
+            getattr(self, 'schema_registry_url', 'http://localhost:8081')
+        )
+        client_config = {'url': schema_registry_url }
+
+        # Use appropriate Schema Registry client based on async mode
+        if is_async:
+            sr_client = AsyncSchemaRegistryClient(client_config)
+        else:
+            sr_client = SchemaRegistryClient(client_config)
+
+        key_serializer = StringSerializer('utf8')
+
+        if serialization_type == 'avro':
+            avro_schema = {
+                "type": "record",
+                "name": "User",
+                "fields": [
+                    {"name": "name", "type": "string"},
+                    {"name": "age", "type": "int"}
+                ]
+            }
+            if is_async:
```

Comment:
read it again and realized I should combine build_serializers and build_async_serializers into one. will refactor